modules/host/network_test.go

```diff
@@ -1,5 +1,61 @@
 package host
 
+import (
+	"testing"
+	"time"
+)
+
+// blockingPortForward is a depedency set that causes the host port froward
+// call at startup to block for 10 seconds, simulating the amount of blocking
+// that can occur in production.
+//
+// blockingPortForward will also cause managedClearPort to always return an
+// error.
+type blockingPortForward struct {
+	productionDependencies
+}
+
+// disrupt will cause the port forward call to block for 10 seconds, but still
+// complete normally. disrupt will also cause managedClearPort to return an
+// error.
+func (blockingPortForward) disrupt(s string) bool {
+	// Return an error when clearing the port.
+	if s == "managedClearPort return error" {
+		return true
+	}
+
+	// Block during port forwarding.
+	if s == "managedForwardPort" {
+		time.Sleep(time.Second * 10)
+	}
+	return false
+}
+
+// TestPortFowardBlocking checks that the host does not accidentally call a
+// write on a closed logger due to a long-running port forward call.
+func TestPortForwardBlocking(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+	ht, err := newMockHostTester(blockingPortForward{}, "TestPortForwardBlocking")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// The close operation would previously fail here because of improper
+	// thread control regarding upnp and shutdown.
+	err = ht.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// The trailing sleep is needed to catch the previously existing error
+	// where the host was not shutting down correctly. Currently, the extra
+	// sleep does nothing, but in the regression a logging panic would occur.
+	time.Sleep(time.Second * 12)
```

Comment:
It needs to outlive the `disrupt` action that causes 10 seconds of sleeping. The `disrupt` action needs live until the host has had enough time to shutdown entirely. I can probably reduce each though.